### PR TITLE
fix: Check if spoke pool is initialized before executing root bundle

### PR DESCRIPTION
**Pull request by nicholaspai**

### CODE REVIEW

File: test/HubPool.Admin.ts [hunk 1]
```diff
@@ -41,8 +41,15 @@ describe("HubPool Admin functions", function () {
   it("Only owner can disable L1 Tokens for liquidity provision", async function () {
```
Comment (nicholaspai) [inline test/HubPool.Admin.ts:41]: Unrelated to PR but these onlyOwner tests are missing

File: contracts/HubPool.sol [hunk 2]
```diff
@@ -554,6 +554,13 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
         // Before interacting with a particular chain's adapter, ensure that the adapter is set.
         require(address(crossChainContracts[poolRebalanceLeaf.chainId].adapter) != address(0), "No adapter for chain");
 
+        // Make sure SpokePool address is initialized since `_sendTokensToChainAndUpdatePooledTokenTrackers` will not
```
Comment (chrismaree) [inline contracts/HubPool.sol:557]: nit: don't use "`".

Comment (nicholaspai) [reply contracts/HubPool.sol:557]: what should I use?

File: contracts/HubPool.sol [hunk 3]
```diff
@@ -779,6 +787,7 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
     // Note this method does a lot and wraps together the sending of tokens and updating the pooled token trackers. This
     // is done as a gas saving so we don't need to iterate over the l1Tokens multiple times.
     function _sendTokensToChainAndUpdatePooledTokenTrackers(
+        address spokePool,
```
Comment (chrismaree) [inline contracts/HubPool.sol:790]: I was going to say that this is redundant butI actually think the way you have it now is cheaper as you only read the state once.

**Review by chrismaree — APPROVED** [review]

**Review by mrice32 — APPROVED** [review]

File: contracts/HubPool.sol (hunk 3 above)
Comment (nicholaspai) [reply contracts/HubPool.sol:790]: exactly

**Pull request merged**